src/pages/ExamsPage/components/AllowanceListActions.jsx

```diff
@@ -10,8 +11,106 @@ import {
 import { DeleteOutline, EditOutline } from '@openedx/paragon/icons';
 import { useIntl } from '@edx/frontend-platform/i18n';
 
+import React, { useState } from 'react';
 import messages from '../messages';
-import { useDeleteAllowance } from '../hooks';
+import { useDeleteAllowance, useEditAllowance, useAllowancesData } from '../hooks';
+
+const EditModal = (isOpen, onClose, allowance, formatMessage) => {
+  const editAllowance = useEditAllowance();
+  const {
+    allowancesList,
+  } = useAllowancesData();
+
+  const allowances = allowancesList.filter(a => a.id === allowance.id);
+  const [additionalTimeError, setAdditionalTimeError] = useState(false);
+
+  const initialFormState = {};
+  const [form, setForm] = useState(initialFormState);
+
+  const handleChange = (event) => {
+    const { name, value } = event.target;
+    setForm(prev => ({ ...prev, [name]: value }));
+  };
+
+  const onSubmit = () => {
+    // todo: make sure that validation is correct
+    setAdditionalTimeError(!form['extra-time-mins']);
+    const valid = (
+      form['extra-time-mins']
+    );
+    if (valid) {
+      form['extra-time-mins'] = Number(form['extra-time-mins']);
+      const payload = {
+        username: allowances[0].username,
+        exam_id: allowance.examId,
+        extra_time_mins: form['extra-time-mins'],
+      };
+      editAllowance(payload, () => {
+        setForm(initialFormState);
+        onClose();
+      });
+    }
+  };
+
+  return (
+    <ModalDialog
+      title="edit allowance"
+      isOpen={isOpen}
+      onClose={onClose}
+      variant="default"
+      hasCloseButton
+      isFullscreenOnMobile
+    >
+      <ModalDialog.Header>
+        <ModalDialog.Title>
+          {formatMessage(messages.editAllowanceHeader)}
+        </ModalDialog.Title>
+      </ModalDialog.Header>
+      <ModalDialog.Body>
+        <Form id="edit-allowance-form" onSubmit={onSubmit}>
```

Comment:
Here is what the edit allowance modal looks like currently (see below). Open to any feedback!

![Screenshot 2024-08-15 at 11 11 42 AM](https://github.com/user-attachments/assets/7f363426-fb3d-41e2-8f11-76f4a1eac9f6)

Reply:
Oh good idea! Let me test/make sure that is handled!

Reply:
Because we are using the form control with a default value and disabled, it looks like this. You can scroll to see the full exam. Does that work okay?

![Screenshot 2024-08-19 at 9 56 33 AM](https://github.com/user-attachments/assets/5c1f80f5-a8dd-4106-8cf9-bb2c564f5282)